llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1208,6 +1232,17 @@ static VSETVLIInfo adjustIncoming(VSETVLIInfo PrevInfo, VSETVLIInfo NewInfo,
 // legal for MI, but may not be the state requested by MI.
 void RISCVInsertVSETVLI::transferBefore(VSETVLIInfo &Info,
                                         const MachineInstr &MI) const {
+  if (isVectorCopy(ST->getRegisterInfo(), MI) &&
+      (Info.isUnknown() || !Info.isValid() || Info.hasSEWLMULRatioOnly())) {
```

Comment:
In principal, we should not need the hasSEWLMULRationOnly case because having such a ratio be constant implies that the VTYPE can't be VILL.  (I think.)  I'm fine with this being a followon change if it ever becomes useful.

Reply:
Agreed, I originally didn't include it but it looks like we always need to clear hasSEWLMULRationOnly after transferBefore if something might end up inserting a vsetvli. Otherwise we trigger an assertion in emitVSETVLIs

Reply:
Understood.